bindings/alicloud/sls/sls.go

```diff
@@ -0,0 +1,140 @@
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/kit/logger"
+)
+
+type AliCloudSlsLogstorage struct {
+	logger   logger.Logger
+	producer *producer.Producer
+	metadata SlsLogstorageMetadata
+}
+
+type SlsLogstorageMetadata struct {
+	Endpoint        string `json:"endpoint"`
+	AccessKeyID     string `json:"accessKeyID"`
+	AccessKeySecret string `json:"accessKeySecret"`
+}
+
+type Callback struct {
+	s *AliCloudSlsLogstorage
+}
+
+// parse metadata field
+func (s *AliCloudSlsLogstorage) Init(metadata bindings.Metadata) error {
+	m, err := s.parseMeta(metadata)
+	if err != nil {
+		return err
+	}
+	s.metadata = *m
+	producerConfig := producer.GetDefaultProducerConfig()
+	// the config properties in the component yaml file
+	producerConfig.Endpoint = m.Endpoint
+	producerConfig.AccessKeyID = m.AccessKeyID
+	producerConfig.AccessKeySecret = m.AccessKeySecret
+	s.producer = producer.InitProducer(producerConfig)
+
+	s.startSlsProducer()
+	return nil
+}
+
+func (s *AliCloudSlsLogstorage) startSlsProducer() {
+	notifyChan := make(chan os.Signal, 1)
+	signal.Notify(notifyChan, syscall.SIGTERM, os.Interrupt)
+	// Start producer instancce
+	s.producer.Start()
+}
+
+func NewAliCloudSlsLogstorage(logger logger.Logger) *AliCloudSlsLogstorage {
+	logger.Debug("initialized Sls log storage binding component")
+	s := &AliCloudSlsLogstorage{
+		logger: logger,
+	}
+	return s
+}
+
+func (s *AliCloudSlsLogstorage) Invoke(ctx context.Context, req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	// verify the metadata property
+	if logProject := req.Metadata["project"]; logProject == "" {
+		return nil, fmt.Errorf("SLS binding error: project property not supplied")
+	}
+	if logstore := req.Metadata["logstore"]; logstore == "" {
+		return nil, fmt.Errorf("SLS binding error: logstore property not supplied")
+	}
+	if topic := req.Metadata["topic"]; topic == "" {
+		return nil, fmt.Errorf("SLS binding error: topic property not supplied")
+	}
+	if source := req.Metadata["source"]; source == "" {
+		return nil, fmt.Errorf("SLS binding error: source property not supplied")
+	}
+
+	log, err := s.parseLog(req)
+	if err != nil {
+		s.logger.Info(err)
+		return nil, err
+	}
+
+	s.logger.Debug(log)
+	callBack := &Callback{}
+	err = s.producer.SendLogWithCallBack(req.Metadata["project"], req.Metadata["logstore"], req.Metadata["topic"], req.Metadata["source"], log, callBack)
+	if err != nil {
+		s.logger.Info(err)
+		return nil, err
+	}
+	return nil, err
+}
+
+// parse the log content
+func (s *AliCloudSlsLogstorage) parseLog(req *bindings.InvokeRequest) (*sls.Log, error) {
+	var logInfo map[string]string
+	err := json.Unmarshal(req.Data, &logInfo)
+	if err != nil {
+		return nil, err
+	}
+	return producer.GenerateLog(uint32(time.Now().Unix()), logInfo), nil
+}
+
+func (s *AliCloudSlsLogstorage) parseMeta(metadata bindings.Metadata) (*SlsLogstorageMetadata, error) {
+	b, err := json.Marshal(metadata.Properties)
```

Comment:
Instead of the two round of JSON, please use `config.Decode`. Example: https://github.com/dapr/components-contrib/blob/d38c7867715fd5d2d0d788fa440b22dd8440e49f/bindings/zeebe/client.go#L72

Reply:
ok, fixed it already